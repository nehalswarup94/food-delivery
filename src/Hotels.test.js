import React from 'react';
import { mount } from 'enzyme'; //test hooks using mount
import toJson from 'enzyme-to-json';
import { act } from 'react-dom/test-utils'; //handles state changes
import configureStore from 'redux-mock-store'; //to use redux store
import * as reactRedux from 'react-redux'; //To use useSelector and useDispatch

describe('My Main component', () => {
    const initialState = { hotels: [] };
    const mockStore = configureStore();
    let store, container;

    const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
    const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');

    jest.mock('./actions/index', () => {
        return {
            __esModule: true,
            default: async () => [
                {
                    name: 'Dominos',
                    img: 'C:/Users/user/Desktop/System Design/food-delivery/public/dominos.png',
                    id: 1,
                    location: 'panchkula',
                    rating: 4,
                    tags: ['italian', 'Cheese', 'Pizza', 'Tacos'],
                    ETA: 15,
                },
                {
                    name: 'Pizza Hut',
                    id: 2,
                    location: 'Chandigarh',
                    rating: 3.5,
                    tags: ['italian', 'Cheese', 'Pizza', 'American'],
                    ETA: 35,
                },
            ]
        }
    })


    beforeEach(() => {
        useSelectorMock.mockClear()
        useDispatchMock.mockClear()
        store = mockStore(initialState)

    });



    it('Mounts App components', async () => {
        const App = require('./App').default;
        const handleSort=jest.fn();
        let hotelsArray = [
            {
                name: 'Dominos',
                img: 'C:/Users/user/Desktop/System Design/food-delivery/public/dominos.png',
                id: 1,
                location: 'panchkula',
                rating: 4,
                tags: ['italian', 'Cheese', 'Pizza', 'Tacos'],
                ETA: 15,
            },
            {
                name: 'Pizza Hut',
                id: 2,
                location: 'Chandigarh',
                rating: 3.5,
                tags: ['italian', 'Cheese', 'Pizza', 'American'],
                ETA: 35,
            },
        ]
        useSelectorMock.mockReturnValue({ hotels: hotelsArray });
        let wrapper;
        await act(async () => {
            wrapper = mount(<App />)
        });

        // wrapper.find('button[value="rating"]').simulate('click');
        // expect(handleSort).toHaveBeenCalled();
        wrapper.update();
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})
