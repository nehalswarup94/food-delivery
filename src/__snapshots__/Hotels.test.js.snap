// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Main component Mounts App components 1`] = `
<App>
  <div
    className="main-div"
  >
    <Search
      handleSearch={[Function]}
    >
      <div
        className="search"
      >
        <input
          className="searchbox"
          name="searchText"
          onChange={[Function]}
          type="text"
          value=""
        />
        <button
          className="search-btn"
          name="search"
          onClick={[Function]}
        >
          Search
        </button>
      </div>
    </Search>
    <Tags
      filterTag={[Function]}
      tags={Array []}
    >
      <div
        className="tags"
      />
    </Tags>
    <div
      className="sort"
    >
      <p>
        Sort accoring to : 
      </p>
      <button
        className="sort-btn"
        onClick={[Function]}
        value="rating"
      >
        Rating
      </button>
      <button
        className="sort-btn"
        onClick={[Function]}
        value="ETA"
      >
        ETA
      </button>
      <button
        className="sort-btn"
        onClick={[Function]}
        value="name"
      >
        Hotel Name
      </button>
    </div>
    <div
      className="hotel-list"
    >
      <Hotels
        filtered={Array []}
        hotels={
          Object {
            "hotels": Array [
              Object {
                "ETA": 15,
                "id": 1,
                "img": "C:/Users/user/Desktop/System Design/food-delivery/public/dominos.png",
                "location": "panchkula",
                "name": "Dominos",
                "rating": 4,
                "tags": Array [
                  "italian",
                  "Cheese",
                  "Pizza",
                  "Tacos",
                ],
              },
              Object {
                "ETA": 35,
                "id": 2,
                "location": "Chandigarh",
                "name": "Pizza Hut",
                "rating": 3.5,
                "tags": Array [
                  "italian",
                  "Cheese",
                  "Pizza",
                  "American",
                ],
              },
            ],
          }
        }
      >
        <div
          className="hotel-card-div"
        />
      </Hotels>
    </div>
  </div>
</App>
`;
